### ISSUE TITLE
State timeline panel: duration value not fixed

### ISSUE THREAD

**Issue by baur** [grafana/grafana#54129]
**What happened**:

Can't see duration value

![state-timeline-duration-error](https://user-images.githubusercontent.com/2898361/186310536-b6afaf3b-0295-442f-9417-c4662b55ab8a.gif)

**What you expected to happen**:

Fixed duration value 

**Environment**:
- Grafana version: v9.1.0
- OS Grafana is installed on: Win10Srv
- User OS & Browser: Chrome

**Comment by usmangt** [comment]
Hi @baur,

Thanks for opening this case.

Can you please try to reproduce this on https://play.grafana.com/ ?

**Comment by baur** [comment]
> Hi @baur,
> 
> Thanks for opening this case.
> 
> Can you please try to reproduce this on https://play.grafana.com/ ?

This is on v9.1.0 
play.grafana.com is v9.0.7 (eed942a50)

Maybe it's related to data?

```CSV
"Time","value"
1661297280000,1
1661297340000,1
1661297400000,1
1661297460000,1
1661297520000,1
1661297580000,1
1661297640000,1
1661297700000,1
1661297760000,1
1661297820000,1
1661297880000,1
1661297940000,1
1661298000000,1
1661298060000,1
1661298120000,1
1661298180000,1
1661298240000,1
1661298300000,1
1661298360000,1
1661298420000,1
1661298480000,1
1661298540000,1
1661298600000,1
1661298660000,1
1661298720000,1
1661298780000,1
1661298840000,1
1661298900000,1
1661298960000,1
1661299020000,1
1661299080000,1
1661299140000,1
1661299200000,1
1661299260000,1
1661299320000,1
1661299380000,1
1661299440000,1
1661299500000,1
1661299560000,1
1661299620000,1
1661299680000,1
1661299740000,1
1661299800000,1
1661299860000,1
1661299920000,1
1661299980000,1
1661300040000,1
1661300100000,1
1661300160000,1
1661300220000,1
1661300280000,1
1661300340000,1
1661300400000,1
1661300460000,1
1661300520000,1
1661300580000,1
1661300640000,1
1661300700000,1
1661300760000,1
1661300820000,1
1661300880000,1
1661300940000,1
1661301000000,1
1661301060000,1
1661301120000,1
1661301180000,1
1661301240000,1
1661301300000,1
1661301360000,1
1661301420000,1
1661301480000,1
1661301540000,0
1661301600000,0
1661301660000,0
1661301720000,0
1661301780000,0
1661301840000,0
1661301900000,1
1661301960000,1
1661302020000,1
1661302080000,1
1661302140000,1
1661302200000,1
1661302260000,1
1661302320000,1
1661302380000,1
1661302440000,1
1661302500000,1
1661302560000,1
1661302620000,1
1661302680000,1
1661302740000,1
1661302800000,1
1661302860000,1
1661302920000,1
1661302980000,1
1661303040000,1
1661303100000,1
1661303160000,1
1661303220000,1
1661303280000,1
1661303340000,1
1661303400000,1
1661303460000,1
1661303520000,1
1661303580000,1
1661303640000,1
1661303700000,1
1661303760000,1
1661303820000,1
1661303880000,1
1661303940000,1
1661304000000,1
1661304060000,1
1661304120000,1
1661304180000,1
1661304240000,1
1661304300000,1
1661304360000,1
1661304420000,1
1661304480000,1
1661304540000,1
1661304600000,1
1661304660000,0
1661304720000,0
1661304780000,0
1661304840000,0
1661304900000,0
1661304960000,0
1661305020000,0
1661305080000,0
1661305140000,0
1661305200000,0
1661305260000,0
1661305320000,0
1661305380000,0
1661305440000,0
1661305500000,0
1661305560000,0
1661305620000,0
1661305680000,0
1661305740000,0
1661305800000,0
1661305860000,0
1661305920000,0
1661305980000,0
1661306040000,0
1661306100000,0
1661306160000,0
1661306220000,0
1661306280000,0
1661306340000,0
1661306400000,0
1661306460000,0
1661306520000,0
1661306580000,0
1661306640000,0
1661306700000,0
1661306760000,0
1661306820000,0
1661306880000,0
1661306940000,0
1661307000000,0
1661307060000,0
1661307120000,0
1661307180000,0
1661307240000,0
1661307300000,0
1661307360000,0
1661307420000,0
1661307480000,0
1661307540000,0
1661307600000,0
1661307660000,0
1661307720000,0
1661307780000,0
1661307840000,0
1661307900000,0
1661307960000,0
1661308020000,0
1661308080000,0
1661308140000,0
1661308200000,0
1661308260000,0
1661308320000,0
1661308380000,0
1661308440000,0
1661308500000,0
1661308560000,0
1661308620000,0
1661308680000,0
1661308740000,0
1661308800000,0
1661308860000,0
1661308920000,0
1661308980000,0
1661309040000,0
1661309100000,0
1661309160000,0
1661309220000,0
1661309280000,0
1661309340000,0
1661309400000,0
1661309460000,0
1661309520000,0
1661309580000,0
1661309640000,0
1661309700000,0
1661309760000,1
1661309820000,1
1661309880000,1
1661309940000,1
1661310000000,1
1661310060000,1
1661310120000,1
1661310180000,1
1661310240000,1
1661310300000,1
1661310360000,1
1661310420000,1
1661310480000,1
1661310540000,1
1661310600000,1
1661310660000,1
1661310720000,1
1661310780000,1
1661310840000,1
1661310900000,1
1661310960000,1
1661311020000,1
```

**Comment by baur** [comment]
also can't hide the tooltip

![state-timeline-duration-error-2](https://user-images.githubusercontent.com/2898361/186384364-6010b0e4-9467-49da-be2f-0e155cc74988.gif)

**Comment by baur** [comment]
also, value mapping is not working 

![image](https://user-images.githubusercontent.com/2898361/186388877-2d691c73-dcfc-4a93-a632-858ca3a8d0e7.png)

**Comment by codeincarnate** [comment]
Thanks for this report @baur! For the main portion of this the "Merge equal consecutive values" option needs to be enabled to show the time duration together otherwise they're considered to be separate segments and so will show the time duration between each event (like you reported).

I was unable to reproduce the issue with value mapping, see below on 9.1.1:

https://user-images.githubusercontent.com/199847/186525432-53ed0102-c85c-4710-9da9-a1efce4bb900.mov

I was able to reproduce the issue with the tooltip. I'll open a separate issue for that.

**Comment by baur** [comment]
>  For the main portion of this the "Merge equal consecutive values" option needs to be enabled to show the time duration together    

![state-timeline-duration-error-3](https://user-images.githubusercontent.com/2898361/186566675-fe1e280c-ff45-41de-b6e5-890d66c5e273.gif)

**Comment by codeincarnate** [comment]
Hmm, it may be that the values are actually different under the state. I'm _think_ we're merging by underlying values here as opposed to the transformed threshold value. It's tricky because for all all the other visualizations there's potentially some differing behavior here. I'll take another look and see if that might be the case here.

**Comment by krzysion** [comment]
From version 9.1,x the option Merge equal consecutive values does not work, example:
https://play.grafana.org/d/8jRyQ13Mz/1-new-features-in-v8-0?orgId=1&viewPanel=65

**Comment by baur** [comment]
> I'm assuming the sample data above is just sample data and not the underlying data from your gif  

This is exactly the kind of data under the state

**Comment by baur** [comment]
> From version 9.1,x the option Merge equal consecutive values does not work, example: https://play.grafana.org/d/8jRyQ13Mz/1-new-features-in-v8-0?orgId=1&viewPanel=65

also not works **Value mapping**

![image](https://user-images.githubusercontent.com/2898361/187817474-7e4b8439-8e8a-4cbc-aa2f-fd58299e94b8.png)

**Comment by mdvictor** [comment]
Work in progress to fix weird duration behaviour

**Comment by baur** [comment]
> Work in progress to fix weird duration behaviour

good news, thank you!
what about **value mapping?**

https://play.grafana.org/d/8jRyQ13Mz/1-new-features-in-v8-0?orgId=1&viewPanel=65

![image](https://user-images.githubusercontent.com/2898361/187817474-7e4b8439-8e8a-4cbc-aa2f-fd58299e94b8.png)

**Comment by mdvictor** [comment]
Hey @baur 

Value mapping still needs to be looked over. But the duration issue will be fixed starting with v9.1.3

**Comment by baur** [comment]
> Hey @baur
> 
> Value mapping still needs to be looked over. But the duration issue will be fixed starting with v9.1.3

#54815
